src/unaudited/EcdsaEpochRegistry_Modular.sol

```diff
@@ -0,0 +1,195 @@
+        uint256 epochZeroValueToSet;
+        if (_epochZeroStart < block.timestamp) {
+            epochZeroValueToSet = block.timestamp;
+        } else {
+            epochZeroValueToSet = _epochZeroStart;
+        }
+        epochZeroStart = epochZeroValueToSet;
+    }
+
+    /*******************************************************************************
+                    EXTERNAL FUNCTIONS - unpermissioned
+    *******************************************************************************/
+    /**
+     * @notice Registers msg.sender as an operator.
+     * @param operatorSignature is the signature of the operator used by the AVS to register the operator in the delegation manager
+     */
+    function registerOperator(
+        ISignatureUtils.SignatureWithSaltAndExpiry memory operatorSignature
+    ) public virtual {
+        address operator = msg.sender;
+        _registerOperator(operator, operatorSignature);
+    }
+
+    // @notice Deregisters the msg.sender.
+    function deregisterOperator() public virtual {
+        address operator = msg.sender;
+        _deregisterOperator(operator);
+    }
+
+    /*******************************************************************************
+                    EXTERNAL FUNCTIONS -- permissioned
+    *******************************************************************************/
+    // @notice TODO: proper documentation
+    function evaluateOperatorSet(
+        address[] memory operatorsToEvaluate,
+        uint256 minimumWeight
+    ) public virtual onlyOwner {
+        _evaluateOperatorSet(operatorsToEvaluate, minimumWeight);
+    }
+
+    /*******************************************************************************
+                            INTERNAL FUNCTIONS
+    *******************************************************************************/
+    // TODO: documentation
+    function _registerOperator(
+        address operator,
+        ISignatureUtils.SignatureWithSaltAndExpiry memory operatorSignature
+    ) internal virtual {
+        if (operatorStatus[operator] != OperatorStatus.REGISTERED) {
+            operatorStatus[operator] = OperatorStatus.REGISTERED;
+
+            // Register the operator with the EigenLayer via this AVS's ServiceManager
+            serviceManager.registerOperatorToAVS(operator, operatorSignature);
+
+            // TODO: event
+            // emit OperatorRegistered(operator, operatorId);
+        }
+    }
+
+    // TODO: documentation
+    function _deregisterOperator(
+        address operator
+    ) internal virtual {
+        if (operatorStatus[operator] == OperatorStatus.REGISTERED) {
+
+            operatorStatus[operator] = OperatorStatus.DEREGISTERED;
+            serviceManager.deregisterOperatorFromAVS(operator);
+
+            // TODO: event
+            // emit OperatorDeregistered(operator, operatorId);
+        }
+    }
+
+    // @notice TODO: proper documentation
+    function _evaluateOperatorSet(
```

Comment:
Wdyt, about a where where operators re-confirm their status as an operator for the upcoming epoch instead of the design where we do updates for the entire operator set. I think this epoch design + operators having a `poke` like function to signal they want to join the next epoch could alleviate some of the updateStakes gas cost issues

Reply:
I think that's OK; as a model in general, requiring a timely `poke` kind of operation from operators seems reasonable.
idk if it's really much better though. kind of just shifts some cost + responsibility to operators, at least in my mind.

Reply:
socializing the cost and responsibility seems beneficial